### PR TITLE
feat(topology): Use Drawer for topology side panel and allow resizing

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**What**: Addresses portions of #5291

### CODE REVIEW

**Review by redallen — APPROVED** [review]
Nice drawers 👖

File: packages/react-integration/cypress/integration/codeeditor.spec.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ describe('Context Selector Demo Test', () => {
       .first()
       .then(code => expect(code).to.have.text('test'));
     cy.get('#reset-code').click();
-    cy.wait(100);
+    cy.wait(1000);
```
Comment (KKoukiou) [inline packages/react-integration/cypress/integration/codeeditor.spec.ts:13]: Why was this necessary timeout increase? Just out of interest!

**Review by KKoukiou — DISMISSED** [review]
Nice cleanup!

**Comment by KKoukiou** [comment]
Just for the record:

Before: 
Screen Shot 2021-03-11 at 09 44 03


After:
Screen Shot 2021-03-11 at 09 41 13

File: packages/react-integration/cypress/integration/codeeditor.spec.ts (hunk 1 above)
Comment (jeff-phillips-18) [reply packages/react-integration/cypress/integration/codeeditor.spec.ts:13]: Tests were failing. This was recently updated to `100` to fix the flakes, but apparently it wasn't enough.

**Comment by jeff-phillips-18** [comment]
@KKoukiou Added a pass thru `onSideBarResize` so applications can track the user desired side bar size for topology.

**Review by KKoukiou — COMMENTED** [review]
In patternfly we rarely encourage the user to pass size/with/height related properties to components. We usually have some preselected sizes, (sm, md, lg) and let the user choose one of them. How do you feel about having preselected sizes here as well?

**Comment by jeff-phillips-18** [comment]
> In patternfly we rarely encourage the user to pass size/with/height related properties to components. We usually have some preselected sizes, (sm, md, lg) and let the user choose one of them. How do you feel about having preselected sizes here as well?

I'm following the `Drawer` parameters here. I think being flexible here is necessary.

**Review by redallen — APPROVED** [review]

**Review by nicolethoen — APPROVED** [review]
Ran it, looks good to me! 
Interesting about needing to up the wait on the code editor test - I feel like the code editor plug in just isn't playing well with the cypress tests... hopefully this fixes the flakiness.
